homeassistant/components/picnic/config_flow.py

```diff
@@ -78,6 +87,31 @@ async def async_step_user(self, user_input=None):
                 step_id="user", data_schema=STEP_USER_DATA_SCHEMA
             )
 
+        return await self._async_picnic_authenticate(
+            "user", STEP_USER_DATA_SCHEMA, user_input
+        )
+
+    async def async_step_reauth(self, config):
```

Comment:
What do you think about sharing `async_step_user` across auth and reauth, and then append CONF_COUNTRY_CODE to the inputs into vol.Schema() depending on whether or not we are in the middle of reauth? I think it might be easier to follow the different states that are possible since it will only be limited to a single function, and no longer have to pass around different step ids, etc.

Reply:
I'll see if that's possible, I though the `step_id` needed to differ for the different flows, but I actually don't know where it's used for.

Reply:
I've merged the steps but called it `authenticate` to make clear it's a common step.